Mathlib/Util/Qq.lean

```diff
@@ -40,4 +40,25 @@ def findLocalDeclWithTypeQ? {u : Level} (sort : Q(Sort u)) : MetaM (Option Q($so
 This is a Qq version of `Lean.Meta.mkDecideProof`. -/
 def mkDecideProofQ (p : Q(Prop)) : MetaM Q($p) := mkDecideProof p
 
+/-- Join a list of elements of type `α` into a container `β`.
+
+Usually `β` is `q(Multiset α)` or `q(Finset α)` or `q(Set α)`.
+
+As an example
+```lean
+mkSetLiteralQ q(Finset ℝ) (List.range 4 |>.map fun n : ℕ ↦ q($n•π))
```

Comment:
Could you also add a test similar to this one, although ideally with no extra imports required?

Reply:
Done - though it's a little less nice than Eric's since it's not using anything noncomputable.